test/test/CMakeLists.txt

```diff
@@ -0,0 +1,78 @@
+file(GLOB TEST_SRCS
+    LIST_DIRECTORIES false
+    CONFIGURE_DEPENDS
+    *.cpp
+)
+list(FILTER TEST_SRCS EXCLUDE REGEX "/(main|pch)\\.cpp")
+
+
+# We can't build test_component[*] for mingw-w64 because it doesn't have an
+# alternative to midl that can produce winmd files. Also, even if we do manage
+# to reuse the MSVC-compiled binaries, mingw-w64 is still missing
+# windowsnumerics.impl.h which is needed to provide the types
```

Comment:
The Clang builds depend on the Windows SDK, just like MSVC. I take it mingw doesn't?

Reply:
They have their own equivalent headers due to the Windows SDK license not allowing redistribution, and the SDK headers depending on too many MSVC-isms

Reply:
It's also important to distinguish key aspect here: Clang can target both MSVC ABI (using Windows SDK) and GCC ABI (using mingw-w64).